### PR TITLE
ci: run `FuzzerSuite` in parallel with more gas

### PR DESCRIPTION
**Pull request by jaschdoc**
Based on https://github.com/flix/flix/pull/9083
Closes https://github.com/flix/flix/issues/9123

### CODE REVIEW

**Comment by jaschdoc** [comment]
I do not understand why `build` is still pending... it was the same issue on the other PR

**Comment by mlutze** [comment]
I think it is because an internal configuration depends on a job called "build", but here you have renamed it to "build and test".
If you name it back, we can do the name change separately

**Comment by jaschdoc** [comment]
> I think it is because an internal configuration depends on a job called "build", but here you have renamed it to "build and test".
> If you name it back, we can do the name change separately 

The same thing happened before I renamed it

**Comment by jaschdoc** [comment]
> I think it is because an internal configuration depends on a job called "build", but here you have renamed it to "build and test".
> If you name it back, we can do the name change separately 

The same thing happened before I renamed it

**Comment by magnus-madsen** [comment]
I cannot even restart it. Its like it does not exist. But perhaps try as Matt suggests and rename it back?

**Comment by jaschdoc** [comment]
Sure :)

**Comment by jaschdoc** [comment]
Oh we found an error in the compiler I think

**Comment by jaschdoc** [comment]
Okay now we are just down to the weird `build` issue and the expected failures from community build and IDE tests

**Comment by jaschdoc** [comment]
Ok I will just split them into separate yaml files

**Comment by jaschdoc** [comment]
If you want to, you can make the fuzzing job required

**Comment by magnus-madsen** [comment]
> If you want to, you can make the fuzzing job required

Lets leave it optional for now.

File: build.gradle [hunk 1]
```diff
@@ -104,6 +104,19 @@ tasks.register('testAll', JavaExec) {
     standardInput = System.in
 }
 
+task testFuzzerSuite(dependsOn: ['testClasses'], type: JavaExec) {
+    mainClass = 'org.scalatest.tools.Runner'
+    // Note: We do not use the -P (parallelism) flag for two reasons:
```
Comment (magnus-madsen) [inline build.gradle:109]: I would drop this comment since this is just a single test suite.

File: build.gradle [hunk 2]
```diff
@@ -104,6 +104,19 @@ tasks.register('testAll', JavaExec) {
...
+    // (A) It may use excessive amounts of memory (running many instances of the compiler).
+    // (B) It may garble the order of the output (when an error occurs).
+    // Of course we still benefit from parallelism inside Flix itself.
+    args = ['-s', 'flix.fuzzers.FuzzerSuite', '-o']
+    // We also enforce that all tests can complete with 4GB of memory:
```
Comment (magnus-madsen) [inline build.gradle:114]: Drop this comment and the memory requirement too.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**